Source/Core/AbsyCmd.cs

```diff
@@ -391,6 +394,30 @@ int FreshAnon()
       }
     }
 
+    void InjectEmptyBigBlockInsideWhileLoopBody(StmtList stmtList) {
+      foreach (var bb in stmtList.BigBlocks) {
+        InjectEmptyBigBlockInsideWhileLoopBody(bb.ec);
+      }
+    }
+    
+    void InjectEmptyBigBlockInsideWhileLoopBody(StructuredCmd structuredCmd) {
+      if (structuredCmd is WhileCmd whileCmd) {
+        InjectEmptyBigBlockInsideWhileLoopBody(whileCmd.Body);
+        if (whileCmd.Body.BigBlocks.Count > 0) {
+          var ec = whileCmd.Body.BigBlocks.Last().ec;
+          if (ec != null && ec is WhileCmd) {
```

Comment:
You can write just `ec is WhileCmd`, since it will true only if if `ec` is non-null.

Reply:
I have simplified the conditional.